src/Device/Driver/AR62xx.cpp

```diff
@@ -0,0 +1,478 @@
+ * but not needed by the driver.
+ * 
+ * ------------------------------------------------------------------------------------------------------
+ * This protocol exists to read/write values in AR/RT/RCU 620xx devices
+ * 
+ * STRUCTURE:
+ * ----------
+ * <HEADER> <PROTOCOL-ID> <FRAME-LENGTH> <FRAME> <CRC>
+ * 
+ * Header:
+ * -------
+ *  is always 0xA5
+ *  length: 1 Byte
+ * 
+ * Protocol-ID:
+ * ------------
+ *  protocol-identifier for the <frame> contents
+ *  The Protocol ID field determines the protocol used for transfer of the package.
+ *  Following protocol ID value is defined:
+ *  0x14 - 62XX Transceiver Family Data Exchange Protocol
+ *  All other values of protocol ID are reserved for different protocols.
+ *  length: 1 Byte
+ * 
+ * Frame-Length:
+ * -------------
+ *  <FRAME> Field length
+ *  length: 1Byte
+ * 
+ * Frame:
+ * ------
+ *  Frame structure:
+ *  <FRAME-ID> <Value-Block>
+ * 
+ *  Frame-ID:
+ *    22: Active / Passive Channel
+ *  
+ *  Frame:
+ *    Active frequency as ID
+ *    Range [0...3039] (118.00 ... 136.990 Mhz) Unit: none (data type: 16-bit unsigned integer)
+ *    Passive frequency as ID
+ *    Range [0...3039] (118.00 ... 136.990 Mhz) Unit: none (data type: 16-bit unsigned integer)
+ *  length: 4 Bytes (Big Endian Order)
+ * 
+ * CRC:
+ * -----
+ *  Cyclic redundancy check
+ *  length: 2 Bytes (Big Endian Order)
+ * 
+ */
+
+#include "Device/Driver/AR62xx.hpp"
+#include "Device/Driver.hpp"
+#include "Device/Port/Port.hpp"
+#include "NMEA/Info.hpp"
+#include "RadioFrequency.hpp"
+#include "LogFile.hpp"
+#include "OS/ByteOrder.hpp"
+
+#include <cstdint>
+
+/* identifier for active / passive station
+ * 1 = active station of AR62xx
+ * 0 = passive station of AR62xx */
+static const unsigned int ACTIVE_STATION = 1;
+static const unsigned int PASSIVE_STATION = 0;
+
+/* list of possible channels which can be set at each frequency
+ * (required for calculation of the frequencyID) */
+static const uint16_t CHANNEL_LIST[] = {0, 5, 10, 15, 25, 30, 35, 40, 50, 55, 60, 65, 75, 80, 85, 90};
+
+/* the amount of channels which can be set at each frequency
+ * (required for calculation of the frequencyID) */
+static const unsigned int NUMBER_OF_CHANNELS = 16;
+
+/* protocol header*/
+static const uint8_t PROTOCOL_HEADER = 0xA5;
+
+/* the protocol id determines the protocol used for transfer of the package.
+the following protocol id defines the 62XX transceiver family data exchange protocol */
+static const uint8_t PROTOCOL_ID = 0x14;
+
+/* The frame-ID to set the active AND passive station
+ *  Frame-ID: 22 */
+static const uint8_t FRAME_ID = 22;
+
+/* The first frequency which can be set in the device 
+ * !!! WITHOUT THE CHANNEL !!!
+ * 
+ * Explanation:
+ * 118000 is the first frequency which can be set in the device
+ * 118.0 is the frequency 00 is the channel (if the device is set to 8khz raster)
+ * */
+static const u_int16_t FIRST_FREQUENCY_WITHOUT_CHANNEL = 1180;
```

Comment:
What is `u_int16_t`?

Reply:
this is the same as uint16_t but changed to uint16_t